staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf.go

```diff
@@ -120,9 +120,7 @@ func (s *Serializer) Decode(originalData []byte, gvk *schema.GroupVersionKind, i
 
 	if intoUnknown, ok := into.(*runtime.Unknown); ok && intoUnknown != nil {
 		*intoUnknown = unk
-		if ok, _, _ := s.RecognizesData(bytes.NewBuffer(unk.Raw)); ok {
-			intoUnknown.ContentType = runtime.ContentTypeProtobuf
-		}
+		intoUnknown.ContentType = runtime.ContentTypeProtobuf
```

Comment:
ContentType might have been set already in the originalData, it should be set here only when not set before, i think.

Reply:
if we're here, `intoUnknown` is pointing at the new `unk` object we created above, which does not have ContentType set